src/main/java/com/amazon/ion/impl/macro/MacroEvaluator.kt

```diff
@@ -87,6 +94,34 @@ class MacroEvaluator {
 
         fun reset() {
             numExpandedExpressions = 0
+            expanderPoolIndex = 0
+            intArrayPoolIndices.fill(0)
+        }
+
+        /** The pool of [IntArray]s for use during this session. */
+        private val intArrayPools = Array<MutableList<IntArray>>(32) { mutableListOf() }
+        private val intArrayPoolIndices = IntArray(32) { 0 }
+
+        /**
+         * Gets an [IntArray] from the pool, or allocates a new one if necessary. The returned array is valid until
+         * [reset] is called.
+         * */
+        fun intArrayForSize(size: Int): IntArray {
+            if (size >= intArrayPools.size) {
+                // Don't attempt to pool arbitrarily large arrays.
+                return IntArray(size)
+            }
```

Comment:
Arrays smaller than `intArrayPools.size` are zeroed out when `reset()` is called, but larger arrays are not. I suppose one could say that the larger vended arrays are still only valid until `reset` is called?

Reply:
`reset()` doesn't zero any of the arrays in the pool, just the indices that point to the next available array in the pool for a given size. Arrays returned from this method are zeroed if freshly allocated, but not otherwise. I'll add a note that the returned array will not necessarily be zeroed.